test/Microsoft.OpenApi.SmokeTests/ApiGurus.cs

```diff
@@ -0,0 +1,107 @@
+    [Collection("DefaultSettings")]
+    public class ApisGuruTests
+    {
+        private static HttpClient _httpClient = new HttpClient(new HttpClientHandler() {
+            AutomaticDecompression = DecompressionMethods.GZip
+        });
+
+        private readonly ITestOutputHelper output;
+    
+        public ApisGuruTests(ITestOutputHelper output)
+        {
+           
+            this.output = output;
+        }
+
+        static ApisGuruTests()
+        {
+            _httpClient.DefaultRequestHeaders.AcceptEncoding.Add(new System.Net.Http.Headers.StringWithQualityHeaderValue("gzip"));
+            _httpClient.DefaultRequestHeaders.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("OpenApi.Net.Tests", "1.0"));
+           
+        }
+
+        public static IEnumerable<object[]> GetSchemas()
+        {
+
+            var listJsonStr = _httpClient
+                    .GetStringAsync("https://api.apis.guru/v2/list.json")
+                    .GetAwaiter().GetResult();
+
+                var json = JObject.Parse(listJsonStr);
+                foreach (var item in json.Properties())
+                {
+                    var versions = GetProp(item.Value, "versions") as JObject;
+                    if (versions == null)
+                        continue;
+                    foreach (var prop in versions.Properties())
+                    {
+                        var urlToJson = GetProp(prop.Value, "swaggerUrl")?.ToObject<string>();
+                        if (urlToJson != null)
+                            yield return new object[] { urlToJson };
+
+                        var utlToYaml = GetProp(prop.Value, "swaggerYamlUrl")?.ToObject<string>();
+                        if (utlToYaml != null)
+                            yield return new object[] { utlToYaml };
+                    }
+            }
+
+            JToken GetProp(JToken obj, string prop)
+            {
+                if (!(obj is JObject jObj))
+                    return null;
+                if (!jObj.TryGetValue(prop, out var jToken))
+                    return null;
+                return jToken;
+            }
+        }
+
+
+        [Theory(DisplayName = "APIs.guru")]
+        [MemberData(nameof(GetSchemas))]
+        public async Task EnsureThatICouldParse(string url)
+        {
+            var stopwatch = new Stopwatch();
+            
+            var response = await _httpClient.GetAsync(url);
+            if (!response.IsSuccessStatusCode)
+            {
+                output.WriteLine($"Couldn't load {url}");
+                return;
+            }
+            await response.Content.LoadIntoBufferAsync();
+            var stream = await response.Content.ReadAsStreamAsync();
+
+            stopwatch.Start();
+
+            var reader = new OpenApiStreamReader(new OpenApiReaderSettings() {
+            });
+            var openApiDocument = reader.Read(stream, out var diagnostic);
+
+            if (diagnostic.Errors.Count > 0)
```

Comment:
same as above, it's a test case, please use the assert clause to test the result.

Reply:
Asserting on the existence of errors in the collection would give erroneous results as many of the OpenAPI descriptions contain spec violations.

Reply:
Hi @darrelmiller - we use multiple validators on the APIs,guru collection, so if this applies to us, we'd certainly appreciate feedback on the spec violations when you have time.

Reply:
@MikeRalphson The two main errors that I am seeing are invalid characters in the component keys and schemas that have a type as an array.  I'll get you a more complete list.  Also, I know many of the Azure documents have `description` properties as peers of a $ref.

Reply:
@darrelmiller many thanks!